src/Core/View.js

```diff
@@ -217,11 +234,14 @@ function _preprocessLayer(view, layer, provider) {
  * });
  *
  * // Add Elevation Layer
- * view.addLayer({
+ * var layer = view.addLayer({
  *      type: 'elevation',
  *      id: 'iElevation',
  * });
  *
+ * // React to preprocessing
+ * layer.whenReady.then(() => { ... });
```

Comment:
If the user didn't pass an instance of Layer to `addLayer` but a simple object with options, we create one for him (in `_preprocessLayer` I think). This instance needs to be returned (that's more or less the same pattern as THREE TextureLoader). 

We could separate layer instanciation from layer addition in the future, or keep it that way, I don't have any strong opinion on the matter for now.